src/pagination/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import i18n from '@dojo/framework/core/middleware/i18n';
+import theme from '@dojo/framework/core/middleware/theme';
+
+import * as css from '../theme/default/pagination.m.css';
+import { WNode } from '@dojo/framework/core/interfaces';
+import Icon from '../icon';
+
+export interface PaginationProperties {
+	initialPage?: number;
+	total: number;
+	onPageChange(page: number): void;
+}
+
+interface PaginationCache {
+	currentPage: number;
+}
+
+const icache = createICacheMiddleware<PaginationCache>();
+const factory = create({ theme, icache, i18n }).properties<PaginationProperties>();
+
+const PAGE_LINKS = 3;
+
+export default factory(function Pagination({ middleware: { theme, icache, i18n }, properties }) {
+	const { initialPage, onPageChange, total } = properties();
+	const classes = theme.classes(css);
+	const currentPage = icache.getOrSet('currentPage', initialPage || 1);
```

Comment:
I believe you'll need to use a `diffProperty` on the `initialPage` value as we do with `initialValue` elsewhere to ensure that it _can_ still be controlled via this property if required.